test/functional/forknotify.py

```diff
@@ -41,11 +42,17 @@ def run_test(self):
         self.nodes[1].generate(1)
         self.sync_all()
 
-        with open(self.alert_filename, 'r', encoding='utf8') as f:
-            alert_text = f.read()
+        # Give bitcoind 10 seconds to write the alert notification
+        timeout = 10
+        while True:
+            with open(self.alert_filename, 'r', encoding='utf8') as f:
+                alert_text = f.read()
+            if len(alert_text) > 0:
+                break
 
-        if len(alert_text) == 0:
-            raise AssertionError("-alertnotify did not warn of up-version blocks")
+            assert timeout > 0, "-alertnotify did not warn of up-version blocks"
```

Comment:
using asserts as control flow is generally an antipattern (I think that the for/else loop is a bit clearer in any case).